### PR TITLE
# [UI] Update ApplicationHeader, InterviewCard, JobInfo, and FormatDate Components

### PR DESCRIPTION
**Pull request by olhasashchuk**
# [UI] Update ApplicationHeader, InterviewCard, JobInfo, and FormatDate Components

---

## Description

This PR addresses UI updates and refactoring in the `ApplicationHeader`, `InterviewCard`, `JobInfo`, and `formatDate` components to improve consistency and readability. Key changes include updates to conditional rendering of location and website, layout adjustments, and a fix to the `formatDate` function for better date formatting.

- **ApplicationHeader**: Adjusted rendering of location and website, ensuring better presentation.
- **InterviewCard**: Minor UI adjustments.
- **JobInfo**: Cleaned up and optimized the date formatting and layout.
- **formatDate**: Refined the date formatting function for improved output.

---

## Changes Made

- **Frontend**:
  - **ApplicationHeader**:
    - Updated the conditional rendering for location and website, with improved fallback UI for missing values.
  - **InterviewCard**:
    - Fixed rendering issues in the date display for interviews.
  - **JobInfo**:
    - Optimized the date display for `applied_date` and `deadline_date`.
    - Cleaned up redundant UI components and improved the use of Material UI components like `Accordion` for better user experience.
  - **formatDate**:
    - Refined the `formatDate` function to provide consistent date formatting across the app.

---

## Screenshots (if applicable)
Screenshot_8
Screenshot_7
---

## Testing Steps

1. **Frontend**:
   - Test the UI rendering in `ApplicationHeader` to ensure proper fallback for location and website when missing.
   - Check the date formatting in `InterviewCard` and `JobInfo` to ensure dates display correctly.
   - Confirm that the application information (location, website, dates) appears as expected across different scenarios (with and without data).
2. **Backend**:
   - No backend changes in this PR.

### CODE REVIEW

File: frontend/app/applications/[id]/ApplicationHeader.js [hunk 1]
```diff
@@ -109,7 +113,12 @@ export default function ApplicationHeader() {
               {application.company_website}
```
Comment (pankaj28843) [inline frontend/app/applications/[id]/ApplicationHeader.js:113]: You have a reusable component for external links, isn't it? Why not us it here?

File: frontend/app/applications/[id]/tabs/JobInfo.js [hunk 2]
```diff
@@ -76,62 +51,94 @@ export default function JobInfo() {
...
+          expandIcon={<ExpandMoreIcon />}
+          aria-controls="job_link-content"
+          id="job_link-header"
+          sx={accordionSummaryStyles}
+        >
+          <Typography sx={{ fontWeight: 600 }}>Job Link</Typography>
+        </AccordionSummary>
+        <AccordionDetails>
+          {application.job_link ? (
+            <Link
```
Comment (pankaj28843) [inline frontend/app/applications/[id]/tabs/JobInfo.js:127]: You have a reusable component for external links, isn't it? Why not us it here?

File: frontend/app/applications/[id]/ApplicationHeader.js (hunk 1 above)
Comment (pankaj28843) [reply frontend/app/applications/[id]/ApplicationHeader.js:113]: See this other comment - https://github.com/FurkannOzbek/application-hub/pull/97/files#r1866723247

File: frontend/app/applications/[id]/tabs/JobInfo.js (hunk 2 above)
Comment (pankaj28843) [reply frontend/app/applications/[id]/tabs/JobInfo.js:127]: See this other comment - https://github.com/FurkannOzbek/application-hub/pull/97/files#r1866723247

Comment (olhasashchuk) [reply frontend/app/applications/[id]/tabs/JobInfo.js]: Thank you! I fixed it!
Screenshot_10

**Review by Natata08 — APPROVED** [review]

**Review by pankaj28843 — APPROVED** [review]

**Pull request merged**